### PR TITLE
Don't try to fetch bundles when disk space is low

### PR DESCRIPTION
**Pull request by vanruch**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: src/workers/atlas_strategies/atlas_challenge_resolution_strategy.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
 
 /** @abstract */
 export default class AtlasChallengeParticipationStrategy {
+  get requiredFreeDiskSpace() {
+    return 50000000; // 50 MB
```
Comment (Anze1m) [inline src/workers/atlas_strategies/atlas_challenge_resolution_strategy.js]: 50MB will be enough? What is the maximum bundle size?

File: src/utils/disk_usage.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import {check} from 'diskusage';
+import os from 'os';
+
+const availableDiskSpace = async () => (await check(os.platform() === 'win32' ? 'c:' : '/')).available;
```
Comment (Anze1m) [inline src/utils/disk_usage.js]: ternary operator 😍

File: src/workers/atlas_strategies/atlas_challenge_resolution_strategy.js (hunk 1 above)
Comment (vanruch) [reply src/workers/atlas_strategies/atlas_challenge_resolution_strategy.js]: That's what strategies are for (not sure if anyone will change it though)

File: src/workers/atlas_worker.js [hunk 3]
```diff
@@ -144,6 +151,18 @@ export default class AtlasWorker extends PeriodicWorker {
     return true;
   }
 
+  async isEnoughAvailableDiskSpace() {
```
Comment (krzysztof-jelski) [inline src/workers/atlas_worker.js]: There's a very visible duplication between `isEnoughAvailableDiskSpace` and `isEnoughFundsToPayForGas`. I think it's worth abstracting.
